awx/main/tasks.py

```diff
@@ -3077,7 +3081,12 @@ def _run_internal(self, receptor_ctl, work_type=None):
                     return res
 
                 if not self.task.instance.result_traceback:
-                    raise RuntimeError(detail)
+                    try:
+                        resultsock = receptor_ctl.get_work_results(self.unit_id, return_sockfile=True)
+                        lines = resultsock.readlines()
+                        self.task.instance.result_traceback = b"".join(lines).decode()
```

Comment:
I'm somewhat confused by the fact that this doesn't raise (which would save the text to `result_traceback` ultimately) and doesn't call `self.task.save(update_fields=['result_traceback'])`. Not saying it doesn't work, it could, but it's not easy to follow.